orbit/cmd/orbit/orbit.go

```diff
@@ -840,3 +851,59 @@ var versionCommand = &cli.Command{
 		return nil
 	},
 }
+
+// capabilitiesChecker is a helper to restart Orbit as soon as certain capabilities
+// are changed in the server.
+//
+// This struct and its methods are designed to play nicely with `oklog.Group`.
+type capabilitiesChecker struct {
+	client        *service.OrbitClient
+	interruptCh   chan struct{} // closed when interrupt is triggered
+	executeDoneCh chan struct{} // closed when execute returns
+}
+
+func newCapabilitiesChecker(client *service.OrbitClient) *capabilitiesChecker {
+	return &capabilitiesChecker{
+		client:        client,
+		interruptCh:   make(chan struct{}),
+		executeDoneCh: make(chan struct{}),
+	}
+}
+
+func (f *capabilitiesChecker) actor() (func() error, func(error)) {
+	return f.execute, f.interrupt
+}
+
+// execute will poll the server for capabilities and emit a stop signal to restart
+// Orbit if certain capabilities are enabled.
+//
+// You need to add an explicit check for each capability you want to watch for
+func (f *capabilitiesChecker) execute() error {
+	defer close(f.executeDoneCh)
+	capabilitiesCHeckTicker := time.NewTicker(5 * time.Minute)
+
+	for {
+		select {
+		case <-capabilitiesCHeckTicker.C:
+			oldCapabilities := f.client.GetServerCapabilities()
```

Comment:
I believe the `capabilitiesChecker` should use its own `orbitClient` (different from the one that `FlagRunner` uses).
Because it may happen that this will return capabilities read from the `FlagRunner` routine.

Reply:
really good catch, thanks! Fixed